### PR TITLE
fix(core): Improve domain and url matching for extractDomain and extractUrl

### PR DESCRIPTION
**Pull request by OlegIvaniv**
This PR includes updates to the regexes used for URL and domain matching to ensure that we handle various edge cases, such as URLs with queries or hashes, localhosts, and URLs with ports. Additionally, I've added documentation that explains the updated regex and included additional test cases to test these edge cases.

Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/workflow/test/ExpressionExtensions/StringExtensions.test.ts [hunk 1]
```diff
@@ -97,13 +97,43 @@ describe('Data Transformation Functions', () => {
 
 		test('.isUrl should work on a string', () => {
 			expect(evaluate('={{ "https://example.com/".isUrl() }}')).toEqual(true);
+			expect(evaluate('={{ "http://example.com/".isUrl() }}')).toEqual(true);
```
Comment (mutdmour) [inline packages/workflow/test/ExpressionExtensions/StringExtensions.test.ts:100]: 😍

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.226.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.226.1)
